apps/frontend/src/app/PageCharacter/CharacterDisplay/Tabs/TabOptimize/Components/UseExcluded.tsx

```diff
@@ -1,23 +1,119 @@
+import useForceUpdate from "../../../../../ReactHooks/useForceUpdate";
+import useMediaQueryUp from "../../../../../ReactHooks/useMediaQueryUp";
+import { filterFunction } from "../../../../../Util/SortByFilters";
 import useBuildSetting from "../useBuildSetting";
 
-export default function UseExcluded({ disabled = false, artsDirty }: { disabled?: boolean, artsDirty: object }) {
+export default function UseExcluded({ disabled = false }: { disabled?: boolean }) {
   const { t } = useTranslation("page_character_optimize")
   const { character: { key: characterKey } } = useContext(CharacterContext)
-  const { buildSetting: { useExcludedArts }, buildSettingDispatch } = useBuildSetting(characterKey)
+  const { buildSetting: { useExcludedArts, artExclusion }, buildSettingDispatch } = useBuildSetting(characterKey)
+  const [show, onOpen, onClose] = useBoolState(false)
+  const numExcludedArt = artExclusion.length
+  const [showSel, onOpenSel, onCloseSel] = useBoolState(false)
+  const onSelect = useCallback((id: string) => {
+    buildSettingDispatch({ artExclusion: [...artExclusion, id] })
+  }, [buildSettingDispatch, artExclusion])
+  const onDelSelect = useCallback((id: string) => {
+    buildSettingDispatch({ artExclusion: artExclusion.filter(i => i !== id) })
+  }, [buildSettingDispatch, artExclusion])
+  return <>
+    <ModalWrapper open={show} onClose={onClose} containerProps={{ maxWidth: "xl" }}><CardDark>
+      <CardContent>
+        <Typography>{t`useExcluded.title`}</Typography>
+        <Divider />
+
+        <ArtifactSelectModal show={showSel} onClose={onCloseSel} onSelect={onSelect} artExclusion={artExclusion} />
+        <Button fullWidth onClick={onOpenSel} color="info" sx={{ mb: 1 }} startIcon={<AddIcon />} >{t`useExcluded.addExc`}</Button>
+        {!!numExcludedArt && <CardLight sx={{ mb: 1 }}><Typography textAlign="center">{t`useExcluded.deSelectExc`}</Typography></CardLight>}
+        <Grid container columns={{ xs: 3, md: 5, lg: 6, xl: 8 }} spacing={1}>
+          {artExclusion.map(id =>
+            <Grid item key={id} xs={1} >
+              <ArtifactCardNano artifactId={id} slotKey="flower" BGComponent={CardLight} onClick={() => onDelSelect(id)} showLocation />
+            </Grid>)}
+        </Grid>
+      </CardContent>
+    </CardDark ></ModalWrapper>
+    <ButtonGroup>
+      <Button fullWidth onClick={() => buildSettingDispatch({ useExcludedArts: !useExcludedArts })} disabled={disabled} startIcon={useExcludedArts ? <CheckBox /> : <CheckBoxOutlineBlank />} color={useExcludedArts ? "success" : "secondary"}>
+        <Box sx={{ display: "flex", gap: 1 }}>
+          <Box><Trans t={t} i18nKey="useExcluded.title" count={numExcludedArt}>Use Excluded Artifacts</Trans></Box>
+          {useExcludedArts ? <SqBadge color="error"><Trans t={t} i18nKey="useExcluded.excNum" count={numExcludedArt}>Excluded <strong>{{ count: numExcludedArt } as TransObject}</strong> artifacts</Trans></SqBadge> :
```

Comment:
Ternary is backwards
![image](https://user-images.githubusercontent.com/36019388/216897180-2e8e771b-f489-47e6-83ae-a2c34c2dbdf2.png)